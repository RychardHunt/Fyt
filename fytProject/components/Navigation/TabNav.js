import React, { Component } from 'react';
import { TabNavigator } from "react-navigation";
import { Button, Text, Icon, Footer, FooterTab } from "native-base";
import Tab1 from '../SampleScreens/Tab1';
import Tab2 from '../SampleScreens/Tab2';
import Tab3 from '../SampleScreens/Tab3';

export default (BottomBar = TabNavigator(
  {
    Tab1: { screen: Tab1 },
    Tab2: { screen: Tab2 },
    Tab3: { screen: Tab3 }
  },
  {
    tabBarPosition: "bottom",
    tabBarComponent: props =>
    {
      return (
        <Footer>
          <FooterTab>
            <Button
              vertical
              active={props.navigationState.index === 0}
              onPress={() => props.navigation.navigate("Tab1")}>
              <Icon name="md-list" />
              <Text>Tab1</Text>
            </Button>
            <Button
              vertical
              active={props.navigationState.index === 1}
              onPress={() => props.navigation.navigate("Tab2")}>
              <Icon name="md-wine" />
              <Text>Tab2</Text>
            </Button>
            <Button
              vertical
              active={props.navigationState.index === 2}
              onPress={() => props.navigation.navigate("Tab3")}>
              <Icon name="md-stopwatch" />
              <Text>Tab3</Text>
            </Button>
          </FooterTab>
        </Footer>
      );
    }
  }
));
