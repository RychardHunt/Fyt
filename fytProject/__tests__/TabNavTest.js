import "react-native";
import React from "react";
import TabNav from "../components/Navigation/TabNav";

import renderer from "react-test-renderer";

test("renders correctly", () => {
  const tree = renderer.create(<TabNav />).toJSON();
  expect(tree).toMatchSnapshot();
});
