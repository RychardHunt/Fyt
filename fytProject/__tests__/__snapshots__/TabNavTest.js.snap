// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  loaded={
    Array [
      0,
    ]
  }
  onLayout={[Function]}
  style={
    Array [
      Object {
        "flex": 1,
        "overflow": "hidden",
      },
      Object {
        "flex": 1,
      },
    ]
  }
>
  <View
    onMoveShouldSetResponder={[Function]}
    onMoveShouldSetResponderCapture={[Function]}
    onResponderEnd={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderReject={[Function]}
    onResponderRelease={[Function]}
    onResponderStart={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    onStartShouldSetResponderCapture={[Function]}
    style={
      Object {
        "alignItems": "stretch",
        "flex": 1,
        "flexDirection": "row",
      }
    }
  >
    <View
      style={
        Object {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
    >
      <View
        collapsable={false}
        removeClippedSubviews={false}
        style={
          Object {
            "flex": 1,
            "overflow": "hidden",
          }
        }
      >
        <View
          style={
            Object {
              "flex": 1,
            }
          }
        >
          <View
            style={
              Array [
                Object {
                  "backgroundColor": "#fff",
                  "flex": 1,
                  "height": 1334,
                },
                Object {
                  "backgroundColor": "#303030",
                  "top": 20,
                },
              ]
            }
          >
            <View
              style={Object {}}
            >
              <View
                onLayout={[Function]}
              >
                <View
                  style={
                    Array [
                      Object {
                        "backgroundColor": "#F8F8F8",
                        "borderBottomColor": "#a7a6ab",
                        "borderBottomWidth": 0.5,
                        "elevation": 3,
                        "flexDirection": "row",
                        "height": 64,
                        "justifyContent": "center",
                        "left": 0,
                        "paddingLeft": 6,
                        "paddingRight": 10,
                        "paddingTop": 18,
                        "right": 0,
                        "shadowColor": undefined,
                        "shadowOffset": undefined,
                        "shadowOpacity": undefined,
                        "shadowRadius": undefined,
                        "top": 0,
                      },
                      Object {
                        "backgroundColor": "#B71C1C",
                      },
                    ]
                  }
                >
                  <View
                    style={
                      Object {
                        "alignItems": "flex-start",
                        "alignSelf": "center",
                        "flex": 1,
                      }
                    }
                  >
                    <View
                      accessible={true}
                      isTVSelectable={true}
                      onResponderGrant={[Function]}
                      onResponderMove={[Function]}
                      onResponderRelease={[Function]}
                      onResponderTerminate={[Function]}
                      onResponderTerminationRequest={[Function]}
                      onStartShouldSetResponder={[Function]}
                      style={
                        Object {
                          "alignItems": "center",
                          "alignSelf": null,
                          "backgroundColor": "transparent",
                          "borderBottomWidth": null,
                          "borderColor": null,
                          "borderLeftWidth": null,
                          "borderRadius": 5,
                          "borderRightWidth": null,
                          "borderTopWidth": null,
                          "borderWidth": undefined,
                          "elevation": 0,
                          "flexDirection": "row",
                          "height": 45,
                          "justifyContent": "space-between",
                          "marginLeft": -3,
                          "opacity": 1,
                          "paddingBottom": 6,
                          "paddingLeft": 4,
                          "paddingRight": 6,
                          "paddingTop": 6,
                          "shadowColor": null,
                          "shadowOffset": null,
                          "shadowOpacity": null,
                          "shadowRadius": null,
                        }
                      }
                    >
                      <Text />
                    </View>
                  </View>
                  <View
                    style={
                      Object {
                        "alignItems": "center",
                        "alignSelf": "center",
                        "flex": 1,
                      }
                    }
                  >
                    <Text
                      numberOfLines={1}
                      style={
                        Object {
                          "color": "#000",
                          "fontFamily": "System",
                          "fontSize": 17,
                          "fontWeight": "700",
                          "marginLeft": undefined,
                          "paddingLeft": 4,
                          "paddingTop": 1,
                          "textAlign": "center",
                        }
                      }
                    >
                      Profile
                    </Text>
                  </View>
                  <View
                    style={
                      Object {
                        "alignItems": "flex-end",
                        "alignSelf": "center",
                        "flex": 1,
                        "flexDirection": "row",
                        "justifyContent": "flex-end",
                      }
                    }
                  />
                </View>
              </View>
            </View>
            <View
              style={
                Object {
                  "alignItems": "center",
                  "flex": 1,
                }
              }
            >
              <RCTScrollView
                contentContainerStyle={
                  Object {
                    "alignItems": "center",
                    "paddingBottom": 50,
                    "width": 712.5,
                  }
                }
                data={
                  Array [
                    "123",
                    "456",
                    "asdf",
                    "asdfaaaaaaaaaaaaaaaaaaaaaaaaaaaaaasdf",
                    "1",
                    "2",
                    "3",
                    "4",
                    "5",
                    "10",
                  ]
                }
                disableVirtualization={false}
                getItem={[Function]}
                getItemCount={[Function]}
                horizontal={false}
                initialNumToRender={10}
                keyExtractor={[Function]}
                maxToRenderPerBatch={10}
                numColumns={2}
                onContentSizeChange={[Function]}
                onEndReachedThreshold={2}
                onLayout={[Function]}
                onMomentumScrollEnd={[Function]}
                onScroll={[Function]}
                onScrollBeginDrag={[Function]}
                onScrollEndDrag={[Function]}
                renderItem={[Function]}
                scrollEventThrottle={50}
                stickyHeaderIndices={Array []}
                updateCellsBatchingPeriod={50}
                viewabilityConfigCallbackPairs={Array []}
                windowSize={21}
              >
                <View>
                  <View
                    onLayout={[Function]}
                    style={null}
                  >
                    <View
                      style={
                        Object {
                          "flexDirection": "row",
                        }
                      }
                    >
                      <View
                        style={
                          Object {
                            "alignItems": "center",
                            "backgroundColor": "lightgray",
                            "borderColor": "black",
                            "borderRadius": 25,
                            "borderWidth": 2,
                            "height": 300,
                            "justifyContent": "center",
                            "margin": 15,
                            "width": 300,
                          }
                        }
                      >
                        <Text
                          style={
                            Array [
                              Object {
                                "color": "#000",
                                "fontFamily": "System",
                                "fontSize": 16,
                              },
                              Object {
                                "fontSize": 20,
                                "padding": 5,
                                "position": "absolute",
                                "top": "10%",
                              },
                            ]
                          }
                          uppercase={false}
                        >
                          123
                        </Text>
                        <View
                          style={
                            Object {
                              "height": 5,
                            }
                          }
                        />
                        <Text />
                      </View>
                      <View
                        style={
                          Object {
                            "alignItems": "center",
                            "backgroundColor": "lightgray",
                            "borderColor": "black",
                            "borderRadius": 25,
                            "borderWidth": 2,
                            "height": 300,
                            "justifyContent": "center",
                            "margin": 15,
                            "width": 300,
                          }
                        }
                      >
                        <Text
                          style={
                            Array [
                              Object {
                                "color": "#000",
                                "fontFamily": "System",
                                "fontSize": 16,
                              },
                              Object {
                                "fontSize": 20,
                                "padding": 5,
                                "position": "absolute",
                                "top": "10%",
                              },
                            ]
                          }
                          uppercase={false}
                        >
                          456
                        </Text>
                        <View
                          style={
                            Object {
                              "height": 5,
                            }
                          }
                        />
                        <Text />
                      </View>
                    </View>
                  </View>
                  <View
                    onLayout={[Function]}
                    style={null}
                  >
                    <View
                      style={
                        Object {
                          "flexDirection": "row",
                        }
                      }
                    >
                      <View
                        style={
                          Object {
                            "alignItems": "center",
                            "backgroundColor": "lightgray",
                            "borderColor": "black",
                            "borderRadius": 25,
                            "borderWidth": 2,
                            "height": 300,
                            "justifyContent": "center",
                            "margin": 15,
                            "width": 300,
                          }
                        }
                      >
                        <Text
                          style={
                            Array [
                              Object {
                                "color": "#000",
                                "fontFamily": "System",
                                "fontSize": 16,
                              },
                              Object {
                                "fontSize": 20,
                                "padding": 5,
                                "position": "absolute",
                                "top": "10%",
                              },
                            ]
                          }
                          uppercase={false}
                        >
                          asdf
                        </Text>
                        <View
                          style={
                            Object {
                              "height": 5,
                            }
                          }
                        />
                        <Text />
                      </View>
                      <View
                        style={
                          Object {
                            "alignItems": "center",
                            "backgroundColor": "lightgray",
                            "borderColor": "black",
                            "borderRadius": 25,
                            "borderWidth": 2,
                            "height": 300,
                            "justifyContent": "center",
                            "margin": 15,
                            "width": 300,
                          }
                        }
                      >
                        <Text
                          style={
                            Array [
                              Object {
                                "color": "#000",
                                "fontFamily": "System",
                                "fontSize": 16,
                              },
                              Object {
                                "fontSize": 20,
                                "padding": 5,
                                "position": "absolute",
                                "top": "10%",
                              },
                            ]
                          }
                          uppercase={false}
                        >
                          asdfaaaaaaaaaaaaaaaaaaaaaaaaaaaaaasdf
                        </Text>
                        <View
                          style={
                            Object {
                              "height": 5,
                            }
                          }
                        />
                        <Text />
                      </View>
                    </View>
                  </View>
                  <View
                    onLayout={[Function]}
                    style={null}
                  >
                    <View
                      style={
                        Object {
                          "flexDirection": "row",
                        }
                      }
                    >
                      <View
                        style={
                          Object {
                            "alignItems": "center",
                            "backgroundColor": "lightgray",
                            "borderColor": "black",
                            "borderRadius": 25,
                            "borderWidth": 2,
                            "height": 300,
                            "justifyContent": "center",
                            "margin": 15,
                            "width": 300,
                          }
                        }
                      >
                        <Text
                          style={
                            Array [
                              Object {
                                "color": "#000",
                                "fontFamily": "System",
                                "fontSize": 16,
                              },
                              Object {
                                "fontSize": 20,
                                "padding": 5,
                                "position": "absolute",
                                "top": "10%",
                              },
                            ]
                          }
                          uppercase={false}
                        >
                          1
                        </Text>
                        <View
                          style={
                            Object {
                              "height": 5,
                            }
                          }
                        />
                        <Text />
                      </View>
                      <View
                        style={
                          Object {
                            "alignItems": "center",
                            "backgroundColor": "lightgray",
                            "borderColor": "black",
                            "borderRadius": 25,
                            "borderWidth": 2,
                            "height": 300,
                            "justifyContent": "center",
                            "margin": 15,
                            "width": 300,
                          }
                        }
                      >
                        <Text
                          style={
                            Array [
                              Object {
                                "color": "#000",
                                "fontFamily": "System",
                                "fontSize": 16,
                              },
                              Object {
                                "fontSize": 20,
                                "padding": 5,
                                "position": "absolute",
                                "top": "10%",
                              },
                            ]
                          }
                          uppercase={false}
                        >
                          2
                        </Text>
                        <View
                          style={
                            Object {
                              "height": 5,
                            }
                          }
                        />
                        <Text />
                      </View>
                    </View>
                  </View>
                  <View
                    onLayout={[Function]}
                    style={null}
                  >
                    <View
                      style={
                        Object {
                          "flexDirection": "row",
                        }
                      }
                    >
                      <View
                        style={
                          Object {
                            "alignItems": "center",
                            "backgroundColor": "lightgray",
                            "borderColor": "black",
                            "borderRadius": 25,
                            "borderWidth": 2,
                            "height": 300,
                            "justifyContent": "center",
                            "margin": 15,
                            "width": 300,
                          }
                        }
                      >
                        <Text
                          style={
                            Array [
                              Object {
                                "color": "#000",
                                "fontFamily": "System",
                                "fontSize": 16,
                              },
                              Object {
                                "fontSize": 20,
                                "padding": 5,
                                "position": "absolute",
                                "top": "10%",
                              },
                            ]
                          }
                          uppercase={false}
                        >
                          3
                        </Text>
                        <View
                          style={
                            Object {
                              "height": 5,
                            }
                          }
                        />
                        <Text />
                      </View>
                      <View
                        style={
                          Object {
                            "alignItems": "center",
                            "backgroundColor": "lightgray",
                            "borderColor": "black",
                            "borderRadius": 25,
                            "borderWidth": 2,
                            "height": 300,
                            "justifyContent": "center",
                            "margin": 15,
                            "width": 300,
                          }
                        }
                      >
                        <Text
                          style={
                            Array [
                              Object {
                                "color": "#000",
                                "fontFamily": "System",
                                "fontSize": 16,
                              },
                              Object {
                                "fontSize": 20,
                                "padding": 5,
                                "position": "absolute",
                                "top": "10%",
                              },
                            ]
                          }
                          uppercase={false}
                        >
                          4
                        </Text>
                        <View
                          style={
                            Object {
                              "height": 5,
                            }
                          }
                        />
                        <Text />
                      </View>
                    </View>
                  </View>
                  <View
                    onLayout={[Function]}
                    style={null}
                  >
                    <View
                      style={
                        Object {
                          "flexDirection": "row",
                        }
                      }
                    >
                      <View
                        style={
                          Object {
                            "alignItems": "center",
                            "backgroundColor": "lightgray",
                            "borderColor": "black",
                            "borderRadius": 25,
                            "borderWidth": 2,
                            "height": 300,
                            "justifyContent": "center",
                            "margin": 15,
                            "width": 300,
                          }
                        }
                      >
                        <Text
                          style={
                            Array [
                              Object {
                                "color": "#000",
                                "fontFamily": "System",
                                "fontSize": 16,
                              },
                              Object {
                                "fontSize": 20,
                                "padding": 5,
                                "position": "absolute",
                                "top": "10%",
                              },
                            ]
                          }
                          uppercase={false}
                        >
                          5
                        </Text>
                        <View
                          style={
                            Object {
                              "height": 5,
                            }
                          }
                        />
                        <Text />
                      </View>
                      <View
                        style={
                          Object {
                            "alignItems": "center",
                            "backgroundColor": "lightgray",
                            "borderColor": "black",
                            "borderRadius": 25,
                            "borderWidth": 2,
                            "height": 300,
                            "justifyContent": "center",
                            "margin": 15,
                            "width": 300,
                          }
                        }
                      >
                        <Text
                          style={
                            Array [
                              Object {
                                "color": "#000",
                                "fontFamily": "System",
                                "fontSize": 16,
                              },
                              Object {
                                "fontSize": 20,
                                "padding": 5,
                                "position": "absolute",
                                "top": "10%",
                              },
                            ]
                          }
                          uppercase={false}
                        >
                          10
                        </Text>
                        <View
                          style={
                            Object {
                              "height": 5,
                            }
                          }
                        />
                        <Text />
                      </View>
                    </View>
                  </View>
                </View>
              </RCTScrollView>
            </View>
          </View>
        </View>
      </View>
    </View>
    <View
      style={null}
    />
    <View
      style={null}
    />
    <View
      style={null}
    />
  </View>
  <View
    style={
      Object {
        "backgroundColor": "#F8F8F8",
        "borderColor": "#cbcbcb",
        "borderTopWidth": 0.5,
        "elevation": 3,
        "flexDirection": "row",
        "height": 55,
        "justifyContent": "center",
        "left": 0,
        "paddingBottom": 0,
        "right": 0,
      }
    }
  >
    <View
      style={
        Object {
          "alignSelf": "stretch",
          "backgroundColor": undefined,
          "flex": 1,
          "flexDirection": "row",
          "justifyContent": "space-between",
        }
      }
    >
      <View
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "alignSelf": "center",
            "backgroundColor": "#cde1f9",
            "borderBottomWidth": null,
            "borderColor": null,
            "borderLeftWidth": null,
            "borderRadius": 5,
            "borderRightWidth": null,
            "borderTopWidth": null,
            "borderWidth": undefined,
            "elevation": 0,
            "flex": 1,
            "flexDirection": "column",
            "height": null,
            "justifyContent": "center",
            "opacity": 1,
            "paddingBottom": 6,
            "paddingTop": 6,
            "shadowColor": null,
            "shadowOffset": null,
            "shadowOpacity": null,
            "shadowRadius": null,
          }
        }
      >
        <Text />
        <Text
          style={
            Object {
              "backgroundColor": "transparent",
              "color": "#007aff",
              "fontFamily": "System",
              "fontSize": 14,
              "lineHeight": 16,
              "marginLeft": 0,
              "marginRight": 0,
              "paddingLeft": 16,
              "paddingRight": 16,
            }
          }
          uppercase={false}
        >
          Profile
        </Text>
      </View>
      <View
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "alignSelf": "center",
            "backgroundColor": "transparent",
            "borderBottomWidth": null,
            "borderColor": null,
            "borderLeftWidth": null,
            "borderRadius": 5,
            "borderRightWidth": null,
            "borderTopWidth": null,
            "borderWidth": undefined,
            "elevation": 0,
            "flex": 1,
            "flexDirection": "column",
            "height": null,
            "justifyContent": "center",
            "opacity": 1,
            "paddingBottom": 6,
            "paddingTop": 6,
            "shadowColor": null,
            "shadowOffset": null,
            "shadowOpacity": null,
            "shadowRadius": null,
          }
        }
      >
        <Text />
        <Text
          style={
            Object {
              "backgroundColor": "transparent",
              "color": "#6b6b6b",
              "fontFamily": "System",
              "fontSize": 14,
              "lineHeight": 16,
              "marginLeft": 0,
              "marginRight": 0,
              "paddingLeft": 16,
              "paddingRight": 16,
            }
          }
          uppercase={false}
        >
          Diet
        </Text>
      </View>
      <View
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "alignSelf": "center",
            "backgroundColor": "transparent",
            "borderBottomWidth": null,
            "borderColor": null,
            "borderLeftWidth": null,
            "borderRadius": 5,
            "borderRightWidth": null,
            "borderTopWidth": null,
            "borderWidth": undefined,
            "elevation": 0,
            "flex": 1,
            "flexDirection": "column",
            "height": null,
            "justifyContent": "center",
            "opacity": 1,
            "paddingBottom": 6,
            "paddingTop": 6,
            "shadowColor": null,
            "shadowOffset": null,
            "shadowOpacity": null,
            "shadowRadius": null,
          }
        }
      >
        <Text />
        <Text
          style={
            Object {
              "backgroundColor": "transparent",
              "color": "#6b6b6b",
              "fontFamily": "System",
              "fontSize": 14,
              "lineHeight": 16,
              "marginLeft": 0,
              "marginRight": 0,
              "paddingLeft": 16,
              "paddingRight": 16,
            }
          }
          uppercase={false}
        >
          Workout
        </Text>
      </View>
      <View
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "alignSelf": "center",
            "backgroundColor": "transparent",
            "borderBottomWidth": null,
            "borderColor": null,
            "borderLeftWidth": null,
            "borderRadius": 5,
            "borderRightWidth": null,
            "borderTopWidth": null,
            "borderWidth": undefined,
            "elevation": 0,
            "flex": 1,
            "flexDirection": "column",
            "height": null,
            "justifyContent": "center",
            "opacity": 1,
            "paddingBottom": 6,
            "paddingTop": 6,
            "shadowColor": null,
            "shadowOffset": null,
            "shadowOpacity": null,
            "shadowRadius": null,
          }
        }
      >
        <Text />
        <Text
          style={
            Object {
              "backgroundColor": "transparent",
              "color": "#6b6b6b",
              "fontFamily": "System",
              "fontSize": 14,
              "lineHeight": 16,
              "marginLeft": 0,
              "marginRight": 0,
              "paddingLeft": 16,
              "paddingRight": 16,
            }
          }
          uppercase={false}
        >
          Playlist
        </Text>
      </View>
    </View>
  </View>
</View>
`;
